### PR TITLE
LightGBM: Restore visibility modifiers.

### PR DESCRIPTION
**Pull request by AlbertoEAF**
Some visibility modifiers were introduced with the port.

This restores the visibility modifiers where appropriate.
This is mainly in the source code, but also includes one of the test schemas that was used for AutoML.

### CODE REVIEW

**Comment by AlbertoEAF** [comment]
@JPDSousa is there an explicit way to mark the default visibility modifier so Codacy does not complain?
Must we change to protected?

File: openml-lightgbm/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainResources.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
     /**
      * Handle for the output parameter necessary for the LightGBM dataset instantiation.
      */
-    public SWIGTYPE_p_p_void swigOutDatasetHandlePtr;
+    SWIGTYPE_p_p_void swigOutDatasetHandlePtr;
```
Comment (JPDSousa) [inline openml-lightgbm/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainResources.java:47]: Why do we need this to be public @AlbertoEAF ?

File: openml-lightgbm/src/test/java/com/feedzai/openml/provider/lightgbm/TestSchemas.java [hunk 2]
```diff
@@ -107,7 +107,7 @@
      * "uuid","cat1_generator","cat2_generator","cat3_generator","num1_float","num2_double","num3_int",
      * "cat1_string","cat2_string","cat3_string"
      */
-    static final DatasetSchema CATEGORICALS_SCHEMA_LABEL_AT_END = new DatasetSchema(
+    public static final DatasetSchema CATEGORICALS_SCHEMA_LABEL_AT_END = new DatasetSchema(
```
Comment (JPDSousa) [inline openml-lightgbm/src/test/java/com/feedzai/openml/provider/lightgbm/TestSchemas.java:110]: Same here: what's the need for the public visibility here?

File: openml-lightgbm/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainResources.java (hunk 1 above)
Comment (shenggwang) [reply openml-lightgbm/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainResources.java:47]: The change is the opposite, we want it to be package-private.
Should we put it as `default`?

Comment (AlbertoEAF) [reply openml-lightgbm/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainResources.java:47]: We don't want the public, we're reverting (remove the 'public'). I don't know how to have it package-private and please codacy. Does that work? `default` as a keyword?

File: openml-lightgbm/src/test/java/com/feedzai/openml/provider/lightgbm/TestSchemas.java (hunk 2 above)
Comment (AlbertoEAF) [reply openml-lightgbm/src/test/java/com/feedzai/openml/provider/lightgbm/TestSchemas.java:110]: I think for the AutoML tests we exposed a schema with numericals and another with numericals + categoricals.

Comment (AlbertoEAF) [reply openml-lightgbm/src/test/java/com/feedzai/openml/provider/lightgbm/TestSchemas.java:110]: This restores that.

File: openml-lightgbm/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainResources.java (hunk 1 above)
Comment (shenggwang) [reply openml-lightgbm/src/main/java/com/feedzai/openml/provider/lightgbm/SWIGTrainResources.java:47]: João Sousa just explained me using this [link](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).
There is no encapsulation for package-private. Therefore, we should ignore codacy, and let it cries as it wants.

**Pull request merged**